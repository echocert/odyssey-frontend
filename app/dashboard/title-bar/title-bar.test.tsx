import { render, screen } from "@testing-library/react";
import "@testing-library/jest-dom";
import userEvent from "@testing-library/user-event";
import { create } from "react-test-renderer";
import TitleBar from "./title-bar";
describe("Title bar", () => {
  it("renders the Title Bar ", () => {
    render(<TitleBar />);

    const title_bar = screen.getByTestId("title-bar");
    expect(title_bar).toBeInTheDocument();
  });
  it("checks the Title Bar login menu is rendered", () => {
    render(<TitleBar />);

    const title_bar_login_menu = screen.getByTestId("title-bar-login-menu");
    userEvent.click(title_bar_login_menu);
  });
  it("checks the Title Bar login notification header is rendered", () => {
    render(<TitleBar />);

    const title_bar_login_notification_header = screen.getByTestId(
      "title-bar-login-menu"
    );
    expect(title_bar_login_notification_header).toBeInTheDocument();
  });

  it("Verify the snapshot of the Title section is matched", () => {
    const wrap = create(<TitleBar />).toJSON();
    expect(wrap).toMatchSnapshot();
  });
});
