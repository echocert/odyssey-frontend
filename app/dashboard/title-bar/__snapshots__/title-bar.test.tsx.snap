// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Title bar Verify the snapshot of the Title section is matched 1`] = `
<div
  className="title-bar"
  data-testid="title-bar"
  id="title-bar"
>
  <div
    className="title-bar__menu-container"
    data-testid="title-bar-login-menu"
  >
    <div
      className="menu"
      data-testid="floating-menu"
    >
      <div
        className="menu__trigger"
        data-testid="floating-menu-trigger"
      >
        <div
          className="menu__anchor"
          data-testid="floating-menu-anchor"
        >
          <div
            data-testid="avatar"
          >
            <button
              className="avatar"
              id="avatar-btn"
            >
              <div
                className="avatar__image"
              />
              <span
                className="avatar__name"
              />
            </button>
          </div>
          <div
            className="menu__content menu__content--left"
            data-testid="floating-menu-content"
            style={
              {
                "--tooltip-left": undefined,
                "display": "",
              }
            }
          >
            <div
              className="menu__content-wrapper"
            >
              <div
                className="title-bar__login-menu"
              >
                <li
                  className="title-bar__login-menu__sub-item"
                >
                  <span
                    className="title-bar__login-menu__sub-item__text"
                  >
                    My Account
                  </span>
                  <div
                    className="title-bar__login-menu__sub-item__line"
                  />
                </li>
                <li
                  className="title-bar__login-menu__sub-item"
                  onClick={[Function]}
                >
                  <span
                    className="title-bar__login-menu__sub-item__text"
                  >
                    Log Out
                  </span>
                </li>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
